### PR TITLE
Delete command added

### LINKED ISSUE #17: Add "delete task" command
Add delete <integer> command

### LINKED ISSUE #29: Add parser for "delete task" command

### PR DESCRIPTION
**Pull request by ivorcmx**
Added "delt" command which supports deleting of task from task list as well as updated parser to support the new command. Fixes issues #29 and #17

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -65,4 +65,22 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+     * e.g. delt 1
+     * Will return false for any other invalid inout
+     * e.g. empty string, delt abc
+     * @throws NullPointerException if {@code s} is null.
+     */
+    public static boolean isInvalidNumber(String s) {
+        requireNonNull(s);
+
+        try {
+            Integer a = Integer.parseInt(s.replaceAll("[^0-9]", ""));
```
Comment (putaojuice) [inline src/main/java/seedu/address/commons/util/StringUtil.java]: Would you consider a mix of alphabets and number as invalid number? E.g. `a2` will become `2` after this replace. However is it intended for the command to work when the user typed `delt a2`?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -121,4 +121,22 @@ public static Tag parseTag(String tag) throws ParseException {
...
+        String trimmedIndex = oneBasedIndex.trim();
+        if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
+            throw new ParseException(MESSAGE_INVALID_INDEX);
+        }
+
+        if (!StringUtil.isInvalidNumber(trimmedIndex)) {
+            throw new ParseException(MESSAGE_INVALID_INDEX);
+        }
+
+        return Integer.parseInt(trimmedIndex.replaceAll("[^0-9]", ""));
```
Comment (putaojuice) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: What if the user typed something like `delt b3`, should the command run or should it prompt an error?

Comment (putaojuice) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Also is it necessary to call `replaceAll()` here? I think when you check for `isNonZeroUnsignedInteger` and `isInvalidNumber`, you already assured that the `trimmedIndex` is a valid number.

**Review by putaojuice — COMMENTED** [review]
Looks good in general, just have a few questions about your implementation in `ParserUtil` and `StringUtil`

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 1 above)
Comment (ivorcmx) [reply src/main/java/seedu/address/commons/util/StringUtil.java]: Fixed with update

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 2 above)
Comment (ivorcmx) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Fixed with update

**Review by putaojuice — APPROVED** [review]
LGTM.

**Pull request merged**